diff --git a/CMake/Common.cmake b/CMake/Common.cmake
index cafe12f..522babe 100644
--- a/CMake/Common.cmake
+++ b/CMake/Common.cmake
@@ -76,7 +76,7 @@ if(APPLE)
 	add_definitions(-DGL_SILENCE_DEPRECATION)
 endif()
 
-set (CMAKE_CXX_STANDARD 11)
+set (CMAKE_CXX_STANDARD 17)
 
 if (MSVC)
 add_definitions(-D_CRT_SECURE_NO_DEPRECATE)
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5b6fbd0..101fcba 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -51,7 +51,7 @@ else()
 	   GIT_REPOSITORY https://github.com/InteractiveComputerGraphics/Discregrid.git
 	   GIT_TAG "ddf20dc0480874bf02e0bdc6ded76c1f101b17fb"
 	   INSTALL_DIR ${ExternalInstallDir}/Discregrid
-	   CMAKE_ARGS -DCMAKE_BUILD_TYPE:STRING=${EXT_CMAKE_BUILD_TYPE} -DCMAKE_INSTALL_PREFIX:PATH=${ExternalInstallDir}/Discregrid
+	   CMAKE_ARGS -DCMAKE_BUILD_TYPE:STRING=${EXT_CMAKE_BUILD_TYPE} -DCMAKE_INSTALL_PREFIX:PATH=${ExternalInstallDir}/Discregrid -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER} -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
 	   -DBUILD_CMD_EXECUTABLE:BOOL=0 -DEIGEN3_INCLUDE_DIR:PATH=${EIGEN3_INCLUDE_DIR} -DCMAKE_CXX_FLAGS=${CMAKE_CXX_FLAGS} -DCMAKE_POLICY_VERSION_MINIMUM=3.10 
 	) 
 	ExternalProject_Get_Property(Ext_Discregrid INSTALL_DIR)
@@ -82,6 +82,10 @@ endif()
 add_subdirectory(PositionBasedDynamics)
 add_subdirectory(Simulation)
 add_subdirectory(Utils)
+if (PBD_LIBS_ONLY AND USE_PYTHON_BINDINGS)
+  add_subdirectory(extern/pybind)
+  add_subdirectory(pyPBD)
+endif()
 if (NOT PBD_LIBS_ONLY)
 	include(DataCopyTargets)
 	add_subdirectory(extern/glfw)
@@ -106,4 +110,3 @@ FILES_MATCHING PATTERN "*.glsl")
 install(DIRECTORY ./Demos/Visualization
 DESTINATION include
 FILES_MATCHING PATTERN "*.h")
-
